### PR TITLE
adding warning when using new-azvmss and not using "latest" image version 

### PR DESCRIPTION
**Pull request by grizzlytheodore**
## Description

adding warning message to use "latest" image version when "latest" is not selected when using new-azvmss command
as per workitem https://msazure.visualstudio.com/One/_queries/edit/5730841/?triage=true

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 [hunk 1]
```diff
@@ -2370,4 +2370,116 @@ function Test-VirtualMachineScaleSetAutoRepair
...
+		If($containsWord -contains $true)
+		{
+			$found = 'true';
+		}
+		else
+		{
+			$found = 'false';  
+		}
+ 
+		Assert-AreEqual 'true' $found;
```
Comment (hyonholee) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:2477]: There is Assert-True

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 [hunk 2]
```diff
@@ -2370,4 +2370,116 @@ function Test-VirtualMachineScaleSetAutoRepair
...
+			-ImageReferencePublisher $imgRef.PublisherName -VhdContainer $vhdContainer `
+			| Add-AzVmssExtension -Name $ExtName -Publisher $Publisher -Type $ExtType -TypeHandlerVersion $ExtVer -AutoUpgradeMinorVersion $True
+ 
+		Start-Transcript -Path "transcript.txt"
+		#create the vmss 
+		New-AzVmss -ResourceGroup $rgname -Name $vmssName -VirtualMachineScaleSet $vmss
+		Stop-Transcript
+ 
+		$wordToFind="You are deploying VMSS pinned to a specific image version from Azure Marketplace.";
+		#$wordToFind ="dummy string";
```
Comment (hyonholee) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:2465]: remove the commented out line

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs [hunk 3]
```diff
@@ -56,7 +56,11 @@ public override void ExecuteCmdlet()
                             string vmScaleSetName = this.VMScaleSetName;
                             VirtualMachineScaleSet parameters = new VirtualMachineScaleSet();
                             ComputeAutomationAutoMapperProfile.Mapper.Map<PSVirtualMachineScaleSet, VirtualMachineScaleSet>(this.VirtualMachineScaleSet, parameters);
-
+                            if (parameters?.VirtualMachineProfile?.StorageProfile?.ImageReference?.Version?.ToLower() != "latest")
+                            {
+                                WriteWarning("You�are�deploying�VMSS�pinned�to�a�specific�image�version�from�Azure�Marketplace.�\n" +
```
Comment (hyonholee) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs:61]: Please also add a message that pinned image version may cause a problem of automatic OS upgrade (or the message from the issue)

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 [hunk 4]
```diff
@@ -2370,4 +2370,116 @@ function Test-VirtualMachineScaleSetAutoRepair
...
+		$file = Get-Content -path "transcript.txt";
+		$containsWord = $file | %{$_ -match $wordToFind}
+		If($containsWord -contains $true)
```
Comment (hyonholee) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:2468]: If there is a line change in the terminal, this test may not work even though the warning message is returned.

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs (hunk 3 above)
Comment (grizzlytheodore) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs:61]: you mean the whole message like this, right? 
You are deploying VMSS pinned to a specific image version from Azure Marketplace. Consider using \"latest\" as the image version.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 (hunk 2 above)
Comment (grizzlytheodore) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1]: removed

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 (hunk 1 above)
Comment (grizzlytheodore) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1]: cleaned up code to use Assert-True instead

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs (hunk 3 above)
Comment (grizzlytheodore) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetCreateOrUpdateMethod.cs]: Message updated to:
"You are deploying VMSS pinned to a specific image version from Azure Marketplace. 
Consider using "latest" as the image version. This allows VMSS to auto upgrade when a newer version is available."

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 (hunk 4 above)
Comment (grizzlytheodore) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1]: updated to join newlines from the transcript files.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 [hunk 5]
```diff
@@ -2370,4 +2370,105 @@ function Test-VirtualMachineScaleSetAutoRepair
...
+			| Add-AzVmssExtension -Name $ExtName -Publisher $Publisher -Type $ExtType -TypeHandlerVersion $ExtVer -AutoUpgradeMinorVersion $True
+ 
+		Start-Transcript -Path "transcript.txt"
+		#create the vmss 
+		New-AzVmss -ResourceGroup $rgname -Name $vmssName -VirtualMachineScaleSet $vmss
+		Stop-Transcript
+ 
+		$wordToFind="You are deploying VMSS pinned to a specific image version from Azure Marketplace.";
+		$file = (Get-Content -path "transcript.txt") -join ' ';
+        Assert-True { $file | %{$_ -match $wordToFind } } ;
```
Comment (hyonholee) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:2466]: align

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 [hunk 6]
```diff
@@ -2370,4 +2370,105 @@ function Test-VirtualMachineScaleSetAutoRepair
...
+#>
+function Test-VirtualMachineScaleSetImageVersion
+{
+	# Setup
+	$rgname = Get-ComputeTestResourceName
+ 
+	try
+	{
+		# Common
+		#$loc = Get-Location "Microsoft.Compute" "virtualMachines";
```
Comment (hyonholee) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:2388]: remove

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 (hunk 5 above)
Comment (hyonholee) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:2466]: Since $file is now a string, you don't need | % part

**Comment by grizzlytheodore** [comment]
@VeryEarly changes have been made, and all checks pass

**Pull request merged**